### PR TITLE
Added country-controller and use auto-complete for country

### LINKED ISSUE #2044: Filter condition country add autocompletion & loader overlap border (screen)
screenshot 2016-02-25 10 13 50

### PR DESCRIPTION
**Pull request by wachterjohannes**
__informations:__

| q                | a
| ---------------- | ---
| Fixed tickets    | fixes #2044 
| Related PRs      | https://github.com/massiveart/husky/pull/609
| BC breaks        | country field has changed see upgrade sql statement
| Documentation PR | none

### CODE REVIEW

**Comment by danrot** [comment]
Does the data have to be migrated?

**Comment by wachterjohannes** [comment]
@danrot i have added the upgrade script.

File: UPGRADE.md [hunk 1]
```diff
@@ -15,6 +15,17 @@ INSERT INTO `re_operators` (`id`, `operator`, `type`, `inputType`) VALUES
     (19, '!=', 6, 'auto-complete');
 ```
 
+Additionally the filter by country has changed. Run following SQL script to update your filter conditions:
+
+```sql
+UPDATE `re_conditions` SET `field` = 'countryId', `type` = 6, `value` = CONCAT('"', (SELECT `id` FROM `co_countries` WHERE `code` = REPLACE(`re_conditions`. `value`, '"', '') LIMIT 1), '"') WHERE `field` = 'countryCode' AND `operator` != 'LIKE';
```
Comment (danrot) [inline UPGRADE.md]: This query does not work for me. The value is written with quotes in the database, which does not happen when I simply add a condition using the new form.

Comment (wachterjohannes) [inline UPGRADE.md]: the values will be `json_encode` by doctrine. if this is not happen for you something is wrong. are you sure that you are in the newest master branch (not on 1.1.9) when testing?

Comment (danrot) [inline UPGRADE.md]: I've created the filters in `1.1.9`. Why should this not work? The projects are probably also updated from this tag (at least that's what should be working anyway).

Comment (wachterjohannes) [inline UPGRADE.md]: because there is another upgrade for the filter until 1.1.9

Comment (wachterjohannes) [inline UPGRADE.md]: https://github.com/sulu-io/sulu/blob/master/UPGRADE.md#filter

**Pull request merged**